### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by ragangithub**
- [link to loom]( https://www.loom.com/share/7244f84a94ce4d88bfb093cfe3c380a5)
- Used ruby gems as software packages system.
- Installed Ruby on Rails framework.
- Understood Rails RESTful design and router.
- Used controllers to handle requests and render empty views.
- Used params from browser request in a safe way.
- Used preprocessed html file with embedded Ruby code.
- Used layouts and templates for shared content.
- Used database migration files to maintain database schema.
- Used validations for models.
- Secure app from n+1 problems.
- Understood what ORM is.
-Wrote SQL queries with ActiveRecord.
- Set up associations between models.
- Used devise gem for authentication.

### CODE REVIEW

File: app/views/payments/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <%= form.label :amount, class: "form-label" %>
+    <%= form.number_field :amount, class: "form-control" %>
+    <% if @transaction.errors.has_key?(:amount) %>
+        <p class="error-message"><%= @transaction.errors[:amount].first %></p>
+      <% end %>
+  </div>
+
+  <div class="field">
+    <%= form.label :categories, class: "form-label" %>
+    <%= form.collection_select :category, @categories, :id, :name, required: true, prompt: true,class: 'form-select', aria_label: "Disabled select example"  %>
```
Comment (bienvenuushindi) [inline app/views/payments/new.html.erb]: :red_circle: [REQUIRED] Using a select field in its current state will only allow the selection of a single category for a payment, which does not fulfil the requirement for a many-to-many association between categories and payments. To enable users to select multiple categories for a payment, you should indeed use either a checkbox field or a select field with multiple options enabled.

Screenshot from 2023-09-13 17-34-24

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
## Hi @ragangithub ,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review and here 👇._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ragangithub** [comment]
- Used a select field with multiple options enabled so that a user can select more than one category for a transaction. Which now fulfills the requirement for a many-to-many association between categories and payments. A user can hold the Ctrl key to select more than one option now

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @ragangithub , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested change implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**